### DISCUSSION TITLE
Using quarto integrated in RStudio

### DISCUSSION THREAD

**Discussion by cbrnr** [quarto-dev/quarto-cli#4440] [Q&A] [answered]
Many people love RStudio's visual editor, especially folks who are not (yet) comfortable with Markdown syntax. I was trying to set up Quarto just to see how easy it is to use for absolute beginners.

According to [this documentation page](https://quarto.org/docs/tools/rstudio.html), it is recommended that people use the integrated Quarto. This is nice, because installing RStudio then offers everything to create Quarto documents.

However, I found two potential issues with such a setup:

1. For generating a PDF, a TeX distribution is required (e.g. TinyTeX). The Job output shows the hint "Please run 'quarto install tinytex' to install TinyTex.". This is (1) misleading because the `quarto` binary included with RStudio is not exposed (and added to `PATH`), and (2) people not familiar with a terminal have no idea what to do. Many people will probably try to install the `tinytex` R package, which does solve the problem (after running `tinytex::install_tinytex()`.
2. I have no idea how to install extensions with the included Quarto. Since the binary is not available on the command line, there doesn't seem to be a way to do this.

I guess my question here (besides tips on how to solve these two issues, especially the second one) is: would it make sense to add the included `quarto` to the `PATH`? Or is there already something that I've overlooked?

Also, I think a setting to tell RStudio which `quarto` to use would be helpful as well, e.g. to switch between the included and an externally installed version.

**Comment by dragonstyle** [comment, marked as answer]
A couple of notes / clarifications:

1. I do think the documentation is mostly explaining to be sure to use a new version of _RStudio_ when working with Quarto, since support for Quarto features is only in newer versions of RStudio.

2. When you install an updated version of Quarto, RStudio should be automatically using that newer version. We prefer the external version because Quarto was so 'young' that we felt it was generally better to always prefer the externally installed version since we knew it would be evolving quickly.

3. Quarto will bind to whatever installation of TeX you have available when producing PDFs (so it will use a TeX installation you've installed for example using TexLive, it will use TinyTex installed using R, or you can use our command line installer to install a bundled version of TinyTex that we have created for that purpose). It's hard for us to recommend something specific to R since Quarto also has first class support for Julia, Python, and other ecosystems. Similarly, installing TeX can be sufficiently daunting that we felt sending users down that path wasn't necessarily the right answer either. Perhaps the path issue you describe is the ultimate solution here (see my next answer).

4. I'm actually not sure why the built in binary isn't on the terminal path in RStudio, it does seem to me like that would make sense, though I'm not sure if I'm missing something important. If you install the updated version, we do add it to the path, so simply updating and restarting RStudio (I think) should make Quarto available in the terminal.

**Reply by cbrnr** [reply]
Thanks for clarifying @dragonstyle, this really helps! You are right, I misread the documentation and thought it recommended the integrated Quarto version. Maybe this could be made even clearer by changing the sentence to something like:

> If you are using Quarto within RStudio it is strongly recommended that you use RStudio v2022.07 or later (the documentation below assumes you are using this build).

It seems like installing Quarto separately is almost always the better option, but if people want to use the RStudio version I think adding it to the path would make sense (at least optionally). I'm not sure how easy that is on macOS, because installing is just dragging the app into the Applications folder (the binary is located in `RStudio.app/Contents/Resources/app/quarto/bin`).

Maybe it would be sufficient to state that if you want to use extensions, you have to install Quarto separately.

Finally, I think it would also be helpful if Quarto stated its version on every run. That way, it would be easy to identify which version is being used (and for the integrated version, it could also say that this is Quarto as shipped with RStudio or something).

**Reply by cscheid** [reply]
> Finally, I think it would also be helpful if Quarto stated its version on every run. 

I _very strongly_ agree with this, for the record.
